library/src/scala/runtime/TupleXXL.scala

```diff
@@ -1,23 +1,41 @@
 package scala.runtime
-import java.util.Arrays.{deepEquals, deepHashCode}
 
 final class TupleXXL private (es: IArray[Object]) extends Product {
   assert(es.length > 22)
 
   def productElement(n: Int): Any = es(n)
   def productArity: Int = es.length
+  override def productPrefix: String = "TupleXXL"
+
+  override def toString: String = elems.asInstanceOf[Array[Object]].mkString("(", ",", ")")
+
+  override def hashCode: Int = {
```

Comment:
For case classes we normally delegate to `scala.runtime.ScalaRunTime._hashCode`, couldn't we do this here too ?

Reply:
Yes, we can.

Though we do not seem to be delegating ti this method for case classes.

Reply:
> Though we do not seem to be delegating ti this method for case classes.

We do: https://github.com/lampepfl/dotty/blob/04042381ff99039884cd85d87e22f1463b8133cf/compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala#L284

Reply:
I mean that I saw some code generated in the hashCode, which I used as a template for the first implementation. Maybe we are optimizing by mistake something that we should not.

Reply:
If there's a primitive type field in the case class we do write it by hand.

Reply:
Then that was the case